### PR TITLE
Add java implementation to time is mooney

### PR DESCRIPTION
**Pull request by nathangong**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-993.mdx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		int[] values = new int[n];
+		for (int i = 0; i < values.length; i++) {
+			values[i] = sc.nextInt();
+		}
+
+		List<List<Integer>> adj = new ArrayList<>();
+		for (int i = 0; i < n; i++) adj.add(new ArrayList<>());
+		for (int i = 0; i < m; i++) {
+			int n1 = sc.nextInt()-1;
+			int n2 = sc.nextInt()-1;
```
Comment (TruaShamu) [inline solutions/usaco-993.mdx:43]:
```suggestion
			int n1 = sc.nextInt() - 1;
			int n2 = sc.nextInt() - 1;
```

Comment (TruaShamu) [reply solutions/usaco-993.mdx:43]: this is optional

**Review by TruaShamu — APPROVED** [review]
looks good to me

File: solutions/usaco-993.mdx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+```java
+import java.io.*;
+import java.util.*;
+
+public class TimeIsMooney {
+	public static void main(String[] args) throws IOException {
+		Scanner sc = new Scanner(new File("time.in"));
+		PrintWriter out = new PrintWriter("time.out");
+
+		// max time that we check (it can be any arbitrary large number)
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:27]: ...but if we made it any magnitude larger it would TLE

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-993.mdx (hunk 2 above)
Comment (nathangong) [reply solutions/usaco-993.mdx:27]: true, i just wanted to specify that the number is arbitrary. is there a better way to word it?

File: solutions/usaco-993.mdx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class TimeIsMooney {
+	public static void main(String[] args) throws IOException {
+		Scanner sc = new Scanner(new File("time.in"));
```
Comment (brad182) [inline solutions/usaco-993.mdx:24]: scanner?

**Review by brad182 — APPROVED** [review]

File: solutions/usaco-993.mdx (hunk 3 above)
Comment (nathangong) [reply solutions/usaco-993.mdx:24]: i'll change it to kattio since everyone hates scanner 😢

Comment (nathangong) [reply solutions/usaco-993.mdx:24]: I'll change it to kattio since everyone hates scanner 😢

Comment (nathangong) [reply solutions/usaco-993.mdx:24]: I'll change it to kattio since everyone hates scanner 😢

File: solutions/usaco-993.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-993.mdx]: just say something like "it wouldn't make sense to check any larger number"
and add a quick multiline comment (2-3 lines) on why it'd be stupid to do so

Comment (nathangong) [reply solutions/usaco-993.mdx]: is what i added sufficient

Comment (nathangong) [reply solutions/usaco-993.mdx]: @SansPapyrus683 ^

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-993.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-993.mdx]: oh sorry lol lemme check

File: solutions/usaco-993.mdx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		List<List<Integer>> adj = new ArrayList<>();
+		for (int i = 0; i < n; i++) adj.add(new ArrayList<>());
+		for (int i = 0; i < m; i++) {
+			int n1 = io.nextInt() - 1;
+			int n2 = io.nextInt() - 1;
+			adj.get(n1).add(n2);
+		}
+
+		// dp[t][i] = max amount of moonies Bessie can make after t days, 
+		// ending at city i
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:48]: make this one line? it's ok if it extends by like 4 or 5 chars

File: solutions/usaco-993.mdx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+		long[][] dp = new long[T_MAX][n];
+		for (int i = 0; i < T_MAX; i++) {
+			Arrays.fill(dp[i], -1);
+		}
+
+		long best = 0;
+		dp[0][0] = 0;
+		for (int t = 0; t < T_MAX - 1; t++) {
+			for (int i = 0; i < n; i++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:57]: comments for these two for loops maybe?

File: solutions/usaco-993.mdx (hunk 4 above)
Comment (nathangong) [reply solutions/usaco-993.mdx:48]: 74 characters

Comment (SansPapyrus683) [reply solutions/usaco-993.mdx:48]: ok i'll just believe you

**Pull request merged**